### PR TITLE
Remove non-code files from release archive

### PR DESCRIPTION
**Pull request by sanmai**
See: #24 

### What and why? 🤔

To avoid problems such as in https://github.com/Automattic/stream-builder/pull/24 let's exclude all non-library files (examples, tests, plumbing) from the final release archive.

### Testing Steps ✍️

`git archive --format=tar HEAD | tar t` should get you a list of included files, which should not include tests and examples.

For a summary, see:

```
$ git archive --format=tar HEAD | tar t | cut -f1 -d/ | sort | uniq
composer.json
config
docs
lib
LICENSE
README.md
```

### You're it! 👑

@Automattic/stream-builders

### CODE REVIEW

**Review by markbiek — APPROVED** [review]
Good idea! LGTM!

**Comment by cyle** [comment]
> final release archive

how is this created? the literal [releases](https://github.com/Automattic/stream-builder/releases) that are generated? if so, we should make it clear somewhere, like the README, that the release doesn't include these things. otherwise, folks may download the latest release, hoping to try out the example, and get confused about why it's not there.

**Comment by sanmai** [comment]
@cyle Good point: I updated the documentation to mention that.

File: docs/StreamBuilder-Beginners-Guide.md [hunk 1]
```diff
@@ -54,7 +54,7 @@ As an example of how to use StreamBuilder, we will implement an endpoint to retr
 
 Let's also assume the returned array is a simple `string[]` which contains trending topics as strings.
 
-We also have an example demo application in the `example/` folder which implements a lot of what we're going to discuss here -- so check that out!
+We also have an example demo application in the `example/` folder which implements a lot of what we're going to discuss here -- so check that out! You must clone the Git repository to work with the example, as released archives do not include example code.
```
Comment (cyle) [inline docs/StreamBuilder-Beginners-Guide.md:57]: awesome. mind adding this somewhere prominent in the `example/README.md` file as well?

Comment (sanmai) [reply docs/StreamBuilder-Beginners-Guide.md:57]: Added in both places :+1:

**Review by cyle — APPROVED** [review]

**Pull request merged**